browser/extensions/brave_tor_client_updater.cc

```diff
@@ -135,14 +135,20 @@ void BraveTorClientUpdater::Register() {
   registered_ = true;
 }
 
+void BraveTorClientUpdater::Unregister() {
+  // We don't call BraveComponent::Unregister here in order to prevent tor
+  // executable component from getting deleted when last tor window closed
+  registered_ = false;
+}
+
 void BraveTorClientUpdater::Cleanup() {
-  // Delete tor binaries if tor is disabled by gpo.
+  // Delete tor binaries if tor is disabled.
```

Comment:
I think unregistering tor component will delete component folder immediately?

Reply:
yes, but it won't work for these scenario:
1. User once opened tor window and get tor executable downloaded
2. User relaunch browser
3. User disable tor in setting

Reply:
Sorry, I don't understand. how unregistering component is related with above scenario?

Reply:
now registration only happens with "Tor Window Opened",  after relaunch we won't register the component until tor window opened

Reply:
I thought deleting directory directly and unregistering component would have same result.

Reply:
After relaunch, if we don't open tor window, then it is in an unregistered state. Then unregister it won't have any effects.

Reply:
https://github.com/brave/brave-core/pull/6081/commits/ec6adc21f3882bd9885877f83844911a35ef5b48#diff-6bf7995a812906ed34d369d71cfa4de3R50
This is where we register tor client updater